src/main/java/example/ExampleWorkflowSteps.java

```diff
@@ -25,9 +31,8 @@
 
     @Override public int startToHeartbeatTimeoutSeconds() { return startToHeartbeatTimeoutSeconds; }
 
-    @Override public int inProgressTimerSeconds() {
-        return inProgressTimerSeconds;
+    @Override public InProgressTimerFunction inProgressTimerSecondsFn() {
+        return new ConstantInProgressTimerFunction(inProgressTimerSeconds);
```

Comment:
I don't think that inProgressTimerSeconds should be an argument to the constructor if EXTRACT_STEP isn't going to use it. It's slightly confusing that way.